test/PrettyPrinting-test.jl

```diff
@@ -103,47 +122,70 @@
    @test latex_string(:(a+-(b+d)*c)) == "a - \\left(b + d\\right) c"
    @test latex_string(:(a+(-b)*c)) == "a - b c"
    @test latex_string(:(a+(-b)*(-c))) == "a + b c"
    @test latex_string(:(a+((-1)*b*c))) == "a - b c"
-   @test latex_string(:((-a)/b)) == "-\\frac{a}{b}"
-   @test latex_string(:(-(a/b))) == "-\\frac{a}{b}"
-   @test latex_string(:(-(a/-b))) == "-\\frac{a}{-b}"
+   @test latex_string(:((-a)//b)) == "-\\frac{a}{b}"
+   @test latex_string(:(-(a//b))) == "-\\frac{a}{b}"
+   @test latex_string(:(-(a//-b))) == "-\\frac{a}{-b}"
    @test latex_string(:((-a)^b)) == "\\left(-a\\right)^{b}"
    @test latex_string(:(a+sqrt(b*c))) == "a + \\sqrt{b c}"
-   @test latex_string(:(ff(-a)*-b)) == "-f f\\left(-a\\right) b"
-   @test latex_string(:(a^(-b/c) - ((-a)*b)*c - ((a*b)*(-c*d)))) ==
+   @test latex_string(:(sqrt(a)^b)) == "\\left(\\sqrt{a}\\right)^{b}"
+   @test latex_string(:(ff(-a)*-b)) == "-\\operatorname{ff}\\left(-a\\right) b"
+   @test latex_string(:(a^(-b//c) - ((-a)*b)*c - ((a*b)*(-c*d)))) ==
                                          "a^{-\\frac{b}{c}} + a b c + a b c d"                             
-   @test latex_string(:((-a/b)*c+d*(e/f)*g^h + (-i)*j/k^l)) ==
+   @test latex_string(:((-a//b)*c+d*(e//f)*g^h + ((-i)*j)//k^l)) ==
                   "-\\frac{a}{b} c + d \\frac{e}{f} g^{h} - \\frac{i j}{k^{l}}"
 
    @test latex_string(:(x^((-1)//2))) == "x^{-\\frac{1}{2}}"
 
    # / has same precedence as * in julia
-   @test latex_string(:(x^(-a*b/c))) == "x^{-\\frac{a b}{c}}"
+   @test latex_string(:(x^(-a*b/c))) == "x^{-a b/c}"
    # // has higher precedence in julia
    @test latex_string(:(x^(-a*b//c))) == "x^{-a \\frac{b}{c}}"
 
-   @test latex_string(:((-a*b/c)^x)) == "\\left(-\\frac{a b}{c}\\right)^{x}"
-   @test latex_string(:((a*b/c)^x)) == "\\left(\\frac{a b}{c}\\right)^{x}"
+   @test latex_string(:((-a*b/c)^x)) == "\\left(-a b/c\\right)^{x}"
+   @test latex_string(:((a*b/c)^x)) == "\\left(a b/c\\right)^{x}"
+   @test latex_string(:((-a*b//c)^x)) == "\\left(-a \\frac{b}{c}\\right)^{x}"
+   @test latex_string(:((a*b//c)^x)) == "\\left(a \\frac{b}{c}\\right)^{x}"
 
    @test latex_string(:(cdot(cdot(-a,b),cdot(c,d)))) ==
              "\\left(-a \\cdot b\\right) \\cdot \\left(c \\cdot d\\right)"
 
-   @test latex_string(:(2*α^2-1*α+1)) == "2 \\alpha^{2} - \\alpha + 1"
+   @test latex_string(:(2*α^2-1*α+1)) == "2 {\\alpha}^{2} - {\\alpha} + 1"
 
    @test latex_string(:([a b; c d])) ==
          "\\left(\\begin{array}{cc}\na & b \\\\\nc & d\n\\end{array}\\right)\n"
 
    @test latex_string(:(if a; b; end;)) isa String
    @test latex_string(1.2) isa String
 
+   function limited_latex_string(x)
+      x = AbstractAlgebra.canonicalize(x)
+      b = IOBuffer()
+      c = IOContext(b, :size_limit => 30)
+      AbstractAlgebra.show_obj(c, MIME("text/latex"), x)
+      return String(take!(b))
+   end
+
+   e = Expr(:call, :+, [:(x^$i) for i in 1:200]...)
+   @test length(limited_latex_string(e)) < 200
```

Comment:
Why not test the actual string?

Reply:
I don't know what the string is.

Reply:
I could tell you what the string is after looking at the output of the function, but it is not predictable what it is exactly.